modules/codegen/src/main/scala/com/twilio/guardrail/generators/Java/DropwizardServerGenerator.scala

```diff
@@ -124,6 +131,112 @@ object DropwizardServerGenerator {
       )
   }
 
+  case class SecurityParameters(routeParameters: List[Parameter],
+                                routeStatements: List[Statement],
+                                handlerParameters: List[Parameter],
+                                handlerArgs: List[Expression])
+
+  type UnknownHttpAuthSchemeHandler = (Operation, String, SecurityScheme, String) => Target[SecurityParameters]
+
+  def emptyUnknownHttpAuthSchemeHandler(operation: Operation,
+                                        schemeName: String,
+                                        securityScheme: SecurityScheme,
+                                        authScheme: String): Target[SecurityParameters] =
+    Target.raiseError(s"HTTP auth scheme $authScheme is not yet supported")
+
+  def generateSecurityParams(operation: Operation,
+                             securitySchemes: Map[String, SecurityScheme],
+                             unknownHttpAuthSchemeHandler: UnknownHttpAuthSchemeHandler): Target[SecurityParameters] =
+    Option(operation.getSecurity).toList
+      .flatMap(_.asScala)
+      .flatTraverse({ requirement =>
+        requirement.asScala.toList.traverse({
+          case (schemeName, _) =>
+            securitySchemes
+              .get(schemeName)
+              .fold(
+                Target.raiseError[SecurityParameters](s"Operation '${operation.getOperationId} references undefined security scheme $schemeName")
+              )({
+                case ApiKeySecurityScheme(name, in, typeName, _) =>
+                  for {
+                    annotationName <- in match {
+                      case SwSecurityScheme.In.QUERY =>
+                        Target.pure("QueryParam")
+                      case SwSecurityScheme.In.HEADER =>
+                        Target.pure("HeaderParam")
+                      case SwSecurityScheme.In.COOKIE =>
+                        Target.raiseError[String]("API Key security schemes stored in cookies are not yet supported")
+                    }
+                    rawParameterType <- typeName.fold(Target.pure(STRING_TYPE))(safeParseClassOrInterfaceType)
+                  } yield {
+                    val parameterType    = if (rawParameterType.isOptional) rawParameterType else optionalType(rawParameterType)
+                    val handlerParameter = new Parameter(util.EnumSet.of(FINAL), parameterType, new SimpleName(name.toCamelCase))
+                    val routeParameter = handlerParameter
+                      .clone()
+                      .addAnnotation(new SingleMemberAnnotationExpr(new Name(annotationName), new StringLiteralExpr(name)))
+                    SecurityParameters(List(routeParameter), List.empty, List(handlerParameter), List(new NameExpr(name.toCamelCase)))
+                  }
+
+                case scheme @ HttpSecurityScheme(authScheme, _) =>
+                  authScheme match {
```

Comment:
case sensitivity?

Reply:
The spec doesn't seem to suggest anything but lowercase is allowed.